### DISCUSSION TITLE
Newly installed Nominatim 4.0.1 on Ubuntu 20.04.3, can't access - Any help?

### DISCUSSION THREAD

**Discussion by wck86** [osm-search/Nominatim#2582] [Q&A] [answered]
Hello everyone. Not too long ago I had used some instructions on one page or another to get Nominatim 3.5 up and running on my OSM tile server. I had a hardware issue and ended up having to rebuild everything, and decided to go with the new release. I followed the delightfully simple instructions on this project page and the instructions on the Nominatim.org site [https://nominatim.org/release-docs/latest/appendix/Install-on-Ubuntu-20/]

After the database build, confirmed by nominatim admin --check-database, I tried to set up apache, but when I navigated to the site, it was a blank white page with only [] in the upper left corner. The OSM tile server works fine, but I couldn't seem to access the Nominatim service. 

When logged in as nominatim, I get the following result from nominatim admin --check-database:
nominatim@myserver:~/nominatim-project$ nominatim admin --check-database
2022-01-16 13:20:15: Using project directory: /srv/nominatim/nominatim-project
2022-01-16 13:20:15: Checking database
Checking database connection ... OK
Checking for placex table ... OK
Checking for placex content ... OK
Checking that tokenizer works ... OK
Checking indexing status ... OK
Checking that database indexes are complete ... OK
Checking that all database indexes are valid ... OK
Checking TIGER external data table. ... not applicable

I'm stuck here. Are there any logs or conf files I can share here to help you discern what's going wrong?

**Comment by mtmail** [comment, marked as answer]
Nominatim 4.0 now defaults to a JSON data output. Are you expecting a HTML page with a search box and map like https://nominatim.openstreetmap.org/ui/search.html ? That got split off into a separate project https://github.com/osm-search/nominatim-ui/

**Reply by wck86** [reply]
I was expecting that, thank you! I'm checking that out now, but it seems a little light on instructions. I'm trying to integrate Nominatim into a leaflet map.
`

<script src="/dist/leaflet.js"></script> <style> html, body { height: 100%; padding: 0; margin: 0; } #map { /* configure the size of the map */ width: 100%; height: 100%; } </style>
<script> // initialize Leaflet var map = L.map('map').setView({lon: 0, lat: 0}, 2);
  // add the OpenStreetMap tiles
  L.tileLayer('/hot/{z}/{x}/{y}.png', {
    maxZoom: 20,
    attribution: '&copy; <a href="https://openstreetmap.org/copyright">OpenStreetMap contributors</a>'
  }).addTo(map);

  // show the scale bar on the lower left corner
  L.control.scale({imperial: true, metric: true}).addTo(map);

  // show a marker on the map
  // L.marker({lon: 0, lat: 0}).bindPopup('The center of the world').addTo(map);
</script>
`
Is installing and setting up the front-end UI necessary for that? I plan to run OSRM too, once I get the address search working again.

**Reply by wck86** [reply]
I think I might have it - thanks again!

**Reply by wck86** [reply]
I have the UI installed and it looks good until I try a search - I get this result: Error fetching data from http://localhost/nominatim/status.php?format=json (TypeError: Failed to fetch) 

Any tips on what the problem might be?

**Reply by mtmail** [reply]
See the Configuration section on https://github.com/osm-search/nominatim-ui/#configuration where you need to put the URL to your Nominatim server into a variable. The default is 'http://localhost/nominatim' Add your servername or IP address and port.

**Reply by wck86** [reply]
Looks like I spoke too soon - I was accidently relying on the public geocoder with "L.Control.geocoder().addTo(map);"
Is the code example below, from [https://www.linuxbabe.com/ubuntu/osm-nominatim-geocoding-server-ubuntu-20-04](url), still applicable to version 4.0.1? 

`     var geocoder = L.Control.Geocoder.nominatim({serviceUrl:'http:/'/nominatim/'});
      if (URLSearchParams && location.search) {
        // parse /?geocoder=nominatim from URL
        var params = new URLSearchParams(location.search);
        var geocoderString = params.get('geocoder');
        if (geocoderString && L.Control.Geocoder[geocoderString]) {
          console.log('Using geocoder', geocoderString);
          geocoder = L.Control.Geocoder[geocoderString]();
        } else if (geocoderString) {
          console.warn('Unsupported geocoder', geocoderString);
        }
      }

      var control = L.Control.geocoder({
        query: 'Type Address here',
        placeholder: 'Search here...',
        geocoder: geocoder,
        position: 'topright'
      }).addTo(map);
      var marker;

      setTimeout(function() {
        control.setQuery('Type Address here');
      }, 12000);`

I get a blank white screen when I load the page. Swapping the top line with "L.Control.geocoder().addTo(map);" makes it work again, but apparently not using my own Nominatim service.

**Reply by mtmail** [reply]
You need to change the `{serviceUrl:'http:/'/nominatim/'}` do point to your server (domain or IP address plus port). The `'` between the two slashes is probably a mistake.

**Reply by wck86** [reply]
That got the map showing up again, with the search bar, but when a search is attempted no results are provided. I really appreciate your prompt and effective responses, but if you're able to jump on a quick Zoom meeting with me I think we might be able to quickly find the problem... I'm thinking it might be in my apache configuration. Are you available to jump on a quick Zoom? I'll post the fix here for the benefit of others. [](https://us04web.zoom.us/j/2468326956?pwd=aHp0cm9lbENvVDl2bGtlR1RGVk12Zz09)

**Reply by mtmail** [reply]
Looking at `https://nominatim.org/release-docs/latest/appendix/Install-on-Ubuntu-20/` does your Nominatim installation return results in your browser at `http://localhost/nominatim/search.php?q=paris` (or since we don't know what parts of the world you imported `http://localhost/nominatim/search.php?format=json`)? If not does the Apache error logfile (Ubuntu: /var/log/apache2/error.log) contain errors? Try replacing 'localhost' with the server's IP address (Ubuntu: run `sudo /sbin/ifconfig`) or full domain name.

**Comment by wck86** [comment]
Got it - everything is working great now. Thanks very much! In case you're curious about my application, I'm setting up a map & routing server for a AREDN mesh. [](https://www.arednmesh.org/aredn-advantage)
The service is intended to run independently of the Internet, without depending on utility provided power, and be able to be cost-effectively and quickly deployed under potentially challenging emergency circumstances by regional volunteers. I'm currently supporting email, text, voice, video, and screen sharing meetings and chat, a Nextcloud file server with integrated collaborative document service, a PBX server and other data services. All are available on the regional emergency data network.

The hardware platform I'm building the map & routing server on is a Compulab FitPC4, which has a broad input voltage range (nominal 12V), so can be run off a battery or car cigarette lighter. It's not a very high-power device, but its passive cooling and ruggedized design makes it a pretty good candidate for this project. It's loaded with 16GB RAM and a 3.6TB Kingston Datacenter SSD. Thanks again for the prompt and effective assistance!

**Reply by cormanttech** [reply]
Hi All,

I am currently working on installing Nominatim OSM on my Ubuntu server 20.04, I already did this documentation https://github.com/philipkozeny/nominatim-docker and got a result like this [] when accessing using website. When I access the  http://localhost:8080/search.php?q=avenue%20 it shows me the location of the country that I imported to the database but it is just text results like this:

![image](https://user-images.githubusercontent.com/55179774/178432316-b3b9e066-284b-43ac-a789-036c91622347.png)

Can I also use this https://github.com/osm-search/nominatim-ui/ to apply UI on my nominatim docker? Thank you!

**Reply by lonvia** [reply]
Duplicate of https://github.com/mediagis/nominatim-docker/discussions/341

**Reply by cormanttech** [reply]
Yes sir,

As I get no response on it. Sorry for the rush as I need to finish it ASAP. Thank you.

**Reply by mtmail** [reply]
> As I get no response on it. Sorry for the rush as I need to finish it ASAP.

Your timeline has no effect on our timeline. We're doing support for free as volunteers. What I've seen so far is a lack of basic understanding how webservers and ports work, Docker vs system environment, localhost vs accessing files from the internet. Screenshots suggest it all runs on a desktop computer. It's not possible to help without guiding you through every step and there have been 12 steps and 20 screenshot across multiple discussions already it seems. It's time consuming to look at your individual setup and doesn't help any other users because it's so specific. That's as far as a pubilc disussion forum goes. As a company you can invest in commercial (paid) support. Or hand over the task to a more experienced person at your company.
